src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -194,13 +195,9 @@ private Task BindArgumentsAsync()
                 {
                     // Do nothing, already set the value.
                 }
-                else if (parameter.ParameterInfo.HasDefaultValue)
+                else
                 {
-                    value = parameter.ParameterInfo.DefaultValue;
-                }
-                else if (parameter.ParameterInfo.ParameterType.IsValueType)
-                {
-                    value = Activator.CreateInstance(parameter.ParameterInfo.ParameterType);
+                    ParameterDefaultValue.TryGetDefaultValue(parameter.ParameterInfo, out value);
```

Comment:
Does this need to account for value types when `TryGetValue` returns `false`?

Reply:
Good catch. Fixed it and added tests now.